### PR TITLE
Implement equality operation for shares

### PR DESCRIPTION
**Pull request by Drake-Eidukas**
I discussed with Hanyun, and she's super bogged down with thesis work, so I implemented some of the refactors suggested from @amiller  in #237

### CODE REVIEW

File: honeybadgermpc/equality.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    x = await asyncio.gather(*[gen_test_bit() for _ in range(security_parameter)])
+
+    # Take the product (this is here the same as the "and") of all
+    # the x'es
+    while len(x) > 1:
+        x_first, x_left = context.ShareArray(x[::2]), context.ShareArray(x[1::2])
+        x = (await (x_first * x_left))._shares
+
+    return await x[0].open() != 0
```
Comment (amiller) [inline honeybadgermpc/equality.py]: just `return x[0]` here. Equality should return a secret share of `[a == b]`, it's up to the caller if they wish to publicly reconstruct it

**Review by amiller — CHANGES_REQUESTED** [review]

File: honeybadgermpc/equality.py (hunk 1 above)
Comment (Drake-Eidukas) [reply honeybadgermpc/equality.py]: Sounds good.

**Comment by Drake-Eidukas** [comment]
@amiller  should be good to go now

**Review by amiller — APPROVED** [review]

**Comment by Drake-Eidukas** [comment]
This resolves one of the three points in #27

**Pull request merged**